### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by joelparkk**

### CODE REVIEW

**Review by hemwatie — APPROVED** [review]

**Review by brkg — CHANGES_REQUESTED** [review]
Delete the console logs please

File: src/component_data/kDataParser.ts [hunk 1]
```diff
@@ -1,78 +1,96 @@
...
+import { kObject } from "../kObjects/kObject";
+import { appendFile } from "fs";
 
 const kObjArray: kObject[] = [];
 
-export function parseData(relevantData: kObjects.anyObject[]): kObject[]
-{
-    for(let i = 0; i < relevantData.length; i++){
+export function parseData(relevantData: kObjects.anyObject[]): kObject[] {
+  // console.log('rel',relevantData)
```
Comment (brkg) [inline src/component_data/kDataParser.ts:8]: here

File: src/component_data/kDataParser.ts [hunk 2]
```diff
@@ -1,78 +1,96 @@
...
+          ele.kind,
+          ele.metadata.name,
+          //.name can be anything need to grab dynamically
+          ele.spec.template.metadata.labels.name
+            ? ele.spec.template.metadata.labels.name
+            : ele.spec.template.metadata.labels.app,
+          ele.spec.replicas ? ele.spec.replicas : 1,
+          newContainer
+        );
+        // console.log('new',newDeployment)
```
Comment (brkg) [inline src/component_data/kDataParser.ts:44]: here

File: src/component_data/kDataParser.ts [hunk 3]
```diff
@@ -1,78 +1,96 @@
...
+          ele.metadata.name,
+          ele.spec.replicas,
+          ele.spec.serviceName,
+          newStatefulContainer,
+          newVolumeClaimTemplates
+        );
+        kObjArray.push(newKStatefulSet);
+        // console.log(kObjArray)
+      } else if (ele.kind === "Service") {
+        // console.log('service',ele)
```
Comment (brkg) [inline src/component_data/kDataParser.ts:75]: here

File: src/component_data/kDataParser.ts [hunk 4]
```diff
@@ -1,78 +1,96 @@
...
+          ele.spec.ports[0].port,
+          ele.spec.ports[0].targetPort,
+          //ele.spec.selector['app.kubernetes.io/name']
+          ele.spec.selector.name
+            ? ele.spec.selector.name
+            : ele.spec.selector.app,
+          ele.spec.type
+        );
+        kObjArray.push(newkSerivce);
+        // console.log(kObjArray)
```
Comment (brkg) [inline src/component_data/kDataParser.ts:89]: here

File: src/components/App.tsx [hunk 5]
```diff
@@ -28,6 +28,7 @@ function App() {
       .then((data: any[]) => {
         // Data will be an array of objects. Each object represents a different YAML file.
         parseData(data);
+        // console.log('data',data)
```
Comment (brkg) [inline src/components/App.tsx:31]: here

File: src/components/views/ClusterView.tsx [hunk 6]
```diff
@@ -137,21 +146,22 @@ function ClusterView(props: any) {
...
-            };
-            let edge2 = {
-              data: {
-                source: `${array[i].label} service`,
-                target: `${array[i].label} invis`,
-                label: "invis",
-              },
-            };
-            relevantData.push(newNode, edge,newNode2, edge2);
+            //invis nodes to force placement
```
Comment (brkg) [inline src/components/views/ClusterView.tsx]: here

File: src/components/views/ClusterView.tsx [hunk 7]
```diff
@@ -188,6 +198,7 @@ function ClusterView(props: any) {
         props.setNamespace(getNamespace(event.target._private.data.id));
         props.setMasterNode(event.target._private.data.id);
         props.setTrigger(true);
+        console.log(event.target._private.data.label);
```
Comment (brkg) [inline src/components/views/ClusterView.tsx:174]: here

File: src/components/views/NodeView.tsx [hunk 8]
```diff
@@ -68,9 +68,11 @@ function NodeView(props: any) {
       }
     }
     for(let i = 0; i < targetNode.replicas; i++){
+      console.log('target',targetNode)
```
Comment (brkg) [inline src/components/views/NodeView.tsx]: here

File: src/components/views/NodeView.tsx [hunk 9]
```diff
@@ -151,7 +156,7 @@ function NodeView(props: any) {
         setTarget(event.target._private.data.label.split(":")[0]);
         setImage(event.target._private.data.id.slice(0,event.target._private.data.id.length - 2));
       }
-      else console.log(event.target._private.data.id)
+      console.log(event.target._private.data.target)
```
Comment (brkg) [inline src/components/views/NodeView.tsx:158]: here

File: src/component_data/kDataParser.ts (hunk 1 above)
Comment (joelparkk) [reply src/component_data/kDataParser.ts:8]: done!

File: src/component_data/kDataParser.ts (hunk 2 above)
Comment (joelparkk) [reply src/component_data/kDataParser.ts:44]: removed

File: src/components/views/ClusterView.tsx (hunk 7 above)
Comment (joelparkk) [reply src/components/views/ClusterView.tsx:174]: console log is for onclick

File: src/components/views/NodeView.tsx (hunk 9 above)
Comment (joelparkk) [reply src/components/views/NodeView.tsx:158]: console log is for onclick

**Review by brkg — APPROVED** [review]

**Pull request merged**